### DISCUSSION TITLE
Add pipe stream support

### DISCUSSION THREAD

**Discussion by ghost** [music-assistant/support#1114] [Feature requests and ideas]
I am currently using Owntone + Shairport Sync with a pipe audio stream to airplay automatically to all speakers automatically. I also use this with Spotify Connect which works flawlessly.

As music assistant 2.0 will feature multiple speaker sync, id like to get rid of Owntone.

Is it possible to integrate the possibility to play a pipe stream with metadata via the local music files?

It would also be great if it would automatically start a pipe stream as soon as music assistant discovers a input from that stream

**Comment by marcelveldt** [comment]
I think this will actually be pretty easy to implement, all the plumbing is already there in MA. I had this idea myself too to attach an actual record player and stream that to various speakers. Nice idea!

**Comment by martinsdan** [comment]
This idea never reached the backlog, I assume.

**Reply by OzGav** [reply]
This would be covered by this request https://github.com/orgs/music-assistant/discussions/1156 and will be prioritised based on votes. So vote away!

**Reply by iVolt1** [reply]
The request that OzGav refers to is about input from hardware like line-in.  This request to add pipe stream support is about input from linux pipes or as they are called in the windows world, named pipes.  A linux pipe input to MA could add a lot of flexibility.

**Comment by OzGav** [comment]
@kosolit We are looking into this.
1. What's your current Shairport Sync pipe configuration? (the exact path and format)
2. Do you want to support other pipe sources besides Shairport Sync?
